Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm

```diff
@@ -59,11 +60,201 @@
+- (void)setPDFPluginBase:(WebKit::PDFPluginBase *)plugin;
+- (void)setPDFDocument:(RetainPtr<PDFDocument>)document;
+- (void)setParent:(NSObject *)parent;
+- (id)accessibilityHitTest:(NSPoint)point;
+- (PDFDocument*)document;
+- (NSObject *)parent;
+- (void)gotoDestination:(PDFDestination *)destination;
+// - (PDFRect) accessibilityConvertBoundsToRootView:(PDFRect)rect withPageIndex:(NSInteger)pageIndex;
+@end
+
+@implementation WKAccessibilityPDFDocumentObject
+
+@synthesize pdfPlugin = _pdfPlugin;
+@synthesize pluginElement = _pluginElement;
+
+- (id)initWithPDFDocument:(RetainPtr<PDFDocument>)document andElement:(WebCore::HTMLPlugInElement *)element
+{
+    if (!(self = [super init]))
+        return nil;
+
+    _pdfDocument = document;
+    _pluginElement = element;
+    return self;
+}
+
+- (void)setPDFPluginBase:(WebKit::PDFPluginBase *)plugin
+{
+    _pdfPlugin = plugin;
+}
+
+- (void)setPDFDocument:(RetainPtr<PDFDocument>)document
+{
+    _pdfDocument = document;
+}
+
+- (BOOL)isAccessibilityElement
+{
+    return YES;
+}
+
+- (id)accessibilityFocusedUIElement
+{
+    if (RefPtr activeAnnotation = _pdfPlugin->activeAnnotation()) {
+        if (WebCore::AXObjectCache* existingCache = _pdfPlugin->axObjectCache()) {
+            if (RefPtr object = existingCache->getOrCreate(activeAnnotation->element()))
+            ALLOW_DEPRECATED_DECLARATIONS_BEGIN
+                return [object->wrapper() accessibilityAttributeValue:@"_AXAssociatedPluginParent"];
+            ALLOW_DEPRECATED_DECLARATIONS_END
+        }
+    }
+    for (id page in [self accessibilityChildren]) {
+        id focusedElement = [page accessibilityFocusedUIElement];
+        if (focusedElement != nil)
+            return focusedElement;
+    }
+    return nil;
+}
+
+- (PDFDocument*)document
+{
+    return _pdfDocument.get();
+}
+
+- (NSArray *)accessibilityVisibleChildren
+{
+    NSMutableArray *visiblePageElements = [NSMutableArray new];
+    for (id page in [self accessibilityChildren]) {
+        id focusedElement = [page accessibilityFocusedUIElement];
+        if (focusedElement != nil)
+            [visiblePageElements addObject:page];
+    }
+    return visiblePageElements;
+}
+
+- (NSObject *)parent
+{
+    RetainPtr<WKAccessibilityPDFDocumentObject> protectedSelf = retainPtr(self);
+    if (!protectedSelf->_parent) {
+        callOnMainRunLoopAndWait([&protectedSelf] {
+            if (auto* axObjectCache = protectedSelf->_pdfPlugin->axObjectCache()) {
+                if (RefPtr pluginAxObject = axObjectCache->getOrCreate(protectedSelf->_pluginElement.get()))
+                    protectedSelf->_parent = pluginAxObject->wrapper();
+            }
+        });
+    }
+    return protectedSelf->_parent.get().get();
+}
+
+- (void)setParent:(NSObject *)parent
+{
+    _parent = parent;
+}
+
+ALLOW_DEPRECATED_IMPLEMENTATIONS_BEGIN
+- (id)accessibilityAttributeValue:(NSString *)attribute {
```

Comment:
Curlies on the next line for ObjC methods (https://webkit.org/code-style-guidelines/)

Reply:
Repeat this comment many times